src/test/java/edu/cornell/kfs/module/cam/document/service/impl/CuAssetSubAccountServiceImplTest.java

```diff
@@ -1,241 +1,224 @@
+        plantDetail = createSourceDetail(TEST_PLANT_ACCOUNT, TEST_SUB_ACCOUNT);
+    }
+
+
+
+    @Test
+    public void testWildcardPatternMatchForBlacklist() throws Exception {
+        setupParameterServiceWithPlantAccountValue(TEST_PATTERN_WILDCARD, EvaluationOperator.DISALLOW);
+        assertPreservationOfRegularDetailOnly(FOR_BLACKLIST);
+    }
+
+    @Test
+    public void testExactPatternMatchForWhitelist() throws Exception {
+        setupParameterServiceWithPlantAccountValue(TEST_PATTERN_REGULAR_EXACT_MATCH, EvaluationOperator.ALLOW);
+        assertPreservationOfRegularDetailOnly(FOR_WHITELIST);
+    }
+
+    @Test
+    public void testExactPatternMatchForBlacklist() throws Exception {
+        setupParameterServiceWithPlantAccountValue(TEST_PATTERN_PLANT_EXACT_MATCH, EvaluationOperator.DISALLOW);
+        assertPreservationOfRegularDetailOnly(FOR_BLACKLIST);
+    }
+
+    @Test
+    public void testWildcardPatternMatchForBothAccountsInWhitelist() throws Exception {
+        setupParameterServiceWithPlantAccountValue(TEST_PATTERN_WILDCARD_BOTH_MATCH, EvaluationOperator.ALLOW);
+        assertPreservationOfBothDetails(FOR_WHITELIST);
+    }
+
+    @Test
+    public void testExactPatternMatchForNeitherAccountInBlacklist() throws Exception {
+        setupParameterServiceWithPlantAccountValue(TEST_PATTERN_NO_MATCH, EvaluationOperator.DISALLOW);
+        assertPreservationOfBothDetails(FOR_BLACKLIST);
+    }
+
+    @Test
+    public void testBothDetailsPreservedIfNoPatternsDefinedInBlacklist() throws Exception {
+        setupParameterServiceWithPlantAccountValue(KFSConstants.EMPTY_STRING, EvaluationOperator.DISALLOW);
+        assertPreservationOfBothDetails(FOR_BLACKLIST);
+    }
+
+    @Test
+    public void testBothDetailsPreservedIfParameterDoesNotExist() throws Exception {
+        setupParameterServiceToReturnNullPlantAccountParameter();
+        assertPreservationOfBothDetails(FOR_BLACKLIST);
+    }
+
+    @Test
+    public void testWildcardPatternMatchForMultiValueBlacklist() throws Exception {
+        setupParameterServiceWithPlantAccountValue(TEST_MULTI_PATTERN, EvaluationOperator.DISALLOW);
+        assertPreservationOfRegularDetailOnly(FOR_BLACKLIST);
+    }
+
+    @Test
+    public void testWildcardPatternMatchForMultiValueWhitelist() throws Exception {
+        setupParameterServiceWithPlantAccountValue(TEST_MULTI_PATTERN_REGULAR_MATCH, EvaluationOperator.ALLOW);
+        assertPreservationOfRegularDetailOnly(FOR_WHITELIST);
+    }
+
+    @Test
+    public void testWildcardPatternMatchForNeitherAccountInMultiValueBlacklist() throws Exception {
+        setupParameterServiceWithPlantAccountValue(TEST_MULTI_PATTERN_NO_MATCH, EvaluationOperator.DISALLOW);
+        assertPreservationOfBothDetails(FOR_BLACKLIST);
+    }
+
+    @Test
+    public void testHandlingOfInvalidPatternSetup() throws Exception {
+        setupParameterServiceWithPlantAccountValue(TEST_BAD_PATTERN, EvaluationOperator.DISALLOW);
+
+        try {
```

Comment:
Just to respond to Bryan's comment since I was debating using ExpectedException in some of my Cynergy PRs:

I believe the ExpectedException rule assumes that throwing the exception will be the last thing that your test method does. If there's currently some validation being done after the exception is thrown, then investigation is needed to determine whether it can truly be done prior to the throwing or if the post-validating is necessary. If the post-throwing checks are still needed, then you must use try/catch behavior.